extension/chrome/settings/modules/my_key_update.ts

```diff
@@ -6,79 +6,96 @@ import { Catch } from '../../../js/common/platform/catch.js';
+    this.acctEmail = acctEmail;
+    this.showKeyUrl = Url.create('my_key.htm', uncheckedUrlParams);
+    this.primariKiPromise = (async () => (await Store.keysGet(acctEmail, [longid]))[0])();
+  }
 
-  $('.action_show_public_key').attr('href', showKeyUrl);
-  const inputPrivateKey = $('.input_private_key');
-  const prvHeaders = Pgp.armor.headers('privateKey');
+  async render() {
+    const primaryKi = await this.primariKiPromise;
+    $('.action_show_public_key').attr('href', this.showKeyUrl);
+    Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);
+    $('.email').text(this.acctEmail);
+    $('.key_words').text(primaryKi.keywords).attr('title', primaryKi.longid);
+    this.inputPrivateKey.attr('placeholder', this.inputPrivateKey.attr('placeholder') + ' (' + primaryKi.longid + ')');
+  }
 
-  const [primaryKi] = await Store.keysGet(acctEmail, [longid]);
-
-  Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);
+  setHandlers() {
+    $('.action_update_private_key').click(this.setHandlerPrevent('double', this.updatePrivateKeyHandler));
+    $('.input_passphrase').keydown(this.setHandler((el, ev) => {
+      if (ev.which === 13) {
+        $('.action_update_private_key').click();
+      }
+    }));
+  }
 
-  $('.email').text(acctEmail);
-  $('.key_words').text(primaryKi.keywords).attr('title', primaryKi.longid);
-  inputPrivateKey.attr('placeholder', inputPrivateKey.attr('placeholder') + ' (' + primaryKi.longid + ')');
+  private async storeUpdatedKeyAndPassphrase(primaryKi: KeyInfo, updatedPrv: OpenPGP.key.Key, updatedPrvPassphrase: string) {
+    const storedPassphrase = await Store.passphraseGet(this.acctEmail, primaryKi.longid, true);
+    await Store.keysAdd(this.acctEmail, updatedPrv.armor());
+    await Store.passphraseSave('local', this.acctEmail, primaryKi.longid, typeof storedPassphrase !== 'undefined' ? updatedPrvPassphrase : undefined);
+    await Store.passphraseSave('session', this.acctEmail, primaryKi.longid, typeof storedPassphrase !== 'undefined' ? undefined : updatedPrvPassphrase);
+    if (await Ui.modal.confirm('Public and private key updated locally.\n\nUpdate public records with new Public Key?')) {
+      try {
+        await Ui.modal.info(await Attester.updatePubkey(primaryKi.longid, updatedPrv.toPublic().armor()));
+      } catch (e) {
+        if (Api.err.isSignificant(e)) {
+          Catch.reportErr(e);
+        }
+        await Ui.modal.error(`Error updating public records:\n\n${Api.err.eli5(e)}\n\n(but local update was successful)`);
+      }
+    }
+    window.location.href = this.showKeyUrl;
+  }
 
-  $('.action_update_private_key').click(Ui.event.prevent('double', async () => {
-    const { keys: [uddatedKey] } = await openpgp.key.readArmored(String(inputPrivateKey.val()));
-    const { keys: [uddatedKeyEncrypted] } = await openpgp.key.readArmored(String(inputPrivateKey.val()));
+  private updatePrivateKeyHandler = async () => {
+    const primaryKi = await this.primariKiPromise;
```

Comment:
Same here, don't work with promises, just get the variable in `render` which is async, then directly access variable.